src/runtime/composables/local/useAuth.ts

```diff
@@ -25,43 +25,74 @@ const signIn: SignInFunc<Credentials, any> = async (credentials, signInOptions,
     params: signInParams ?? {}
   })
 
+  const { rawToken, rawRefreshToken } = useAuthState()
+
+  // Extract the access token
   const extractedToken = jsonPointerGet(response, config.token.signInResponseTokenPointer)
   if (typeof extractedToken !== 'string') {
-    console.error(`Auth: string token expected, received instead: ${JSON.stringify(extractedToken)}. Tried to find token at ${config.token.signInResponseTokenPointer} in ${JSON.stringify(response)}`)
+    console.error(
+      `Auth: string token expected, received instead: ${JSON.stringify(extractedToken)}. ` +
+       `Tried to find token at ${config.token.signInResponseTokenPointer} in ${JSON.stringify(response)}`
+    )
     return
   }
-
-  const { rawToken } = useAuthState()
   rawToken.value = extractedToken
 
+  // Extract the refresh token if enabled
+  if (config.refresh.isEnabled) {
+    const refreshTokenPointer = config.refresh.token.signInResponseRefreshTokenPointer
+
+    const extractedRefreshToken = jsonPointerGet(response, refreshTokenPointer)
+    if (typeof extractedRefreshToken !== 'string') {
+      console.error(
+        `Auth: string token expected, received instead: ${JSON.stringify(extractedRefreshToken)}. ` +
+         `Tried to find refresh token at ${refreshTokenPointer} in ${JSON.stringify(response)}`
+      )
+      return
+    }
+    rawRefreshToken.value = extractedRefreshToken
+  }
+
   await nextTick(getSession)
 
-  const { redirect = true } = signInOptions ?? {}
+  const { redirect = true, external } = signInOptions ?? {}
   let { callbackUrl } = signInOptions ?? {}
   if (typeof callbackUrl === 'undefined') {
     callbackUrl = await determineCallbackUrl(runtimeConfig.public.auth, () => getRequestURLWN(nuxt))
   }
   if (redirect) {
-    return navigateTo(callbackUrl)
+    return navigateTo(callbackUrl, { external })
   }
 }
 
 const signOut: SignOutFunc = async (signOutOptions) => {
   const nuxt = useNuxtApp()
   const runtimeConfig = await callWithNuxt(nuxt, useRuntimeConfig)
   const config = useTypedBackendConfig(runtimeConfig, 'local')
-  const { data, rawToken, token } = await callWithNuxt(nuxt, useAuthState)
+  const { data, token, rawToken, refreshToken, rawRefreshToken }: UseAuthStateReturn = await callWithNuxt(nuxt, useAuthState)
+
+  const signOutConfig = config.endpoints.signOut
+
+  let headers
+  let body
+  if (signOutConfig) {
+    headers = new Headers({ [config.token.headerName]: token.value } as HeadersInit)
+    // If the refresh provider is used and the signOut request is of post, include the refreshToken in the body
+    if (config.refresh.isEnabled && signOutConfig.method.toLowerCase() === 'post') {
+      body = {
+        refreshToken: refreshToken.value
+      }
+    }
+  }
```

Comment:
@zoey-kaiser I don't understand where this comes from, it doesn't seem to be in line with other code

Reply:
This was something you had requested in a review on https://github.com/sidebase/nuxt-auth/pull/731#discussion_r1600595494, I had decided to merge the PR regardless, as the author did not respond. I applied the changes you had requested to this PR. 

If you had another implementation in mind, we can also adapt this solution 😊